sql-odbc/src/odfesqlodbc/odbcapi.c

```diff
@@ -1367,4 +1367,33 @@ SQLRETURN SQL_API SQLColAttributes(SQLHSTMT StatementHandle,
     LEAVE_STMT_CS(stmt);
     return ret;
 }
+
+/*	SQLError -> SQLDiagRec */
+RETCODE SQL_API SQLError(SQLHENV EnvironmentHandle, SQLHDBC ConnectionHandle,
+                         SQLHSTMT StatementHandle, SQLCHAR *Sqlstate,
+                         SQLINTEGER *NativeError, SQLCHAR *MessageText,
+                         SQLSMALLINT BufferLength, SQLSMALLINT *TextLength) {
+    RETCODE ret;
+    SQLSMALLINT RecNumber = 1;
+
+    MYLOG(ES_TRACE, "entering\n");
+
+    if (EnvironmentHandle) {
```

Comment:
You should check this Statement, connection, then environment rather than the other way around. The most recently constructed handle gets priority.

Reply:
Reversed order of handle checks